wpilibc/src/main/native/include/frc/smartdashboard/MechanismObject2d.h

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#pragma once
+
+#include <map>
+#include <memory>
+
+#include <wpi/StringMap.h>
+
+#include <networktables/NetworkTable.h>
+
+namespace frc {
+
+class MechanismObject2d {
+  friend class Mechanism2d;
+ protected:
+  explicit MechanismObject2d(const wpi::Twine& name);
+  /**
+   * Update all entries with new ones from a new table.
+   *
+   * @param table the new table.
+   */
+  virtual void UpdateEntries(std::shared_ptr<NetworkTable> table) = 0;
+ public:
+  /**
+   * Retrieve the object's name.
+   * 
+   * @return the object's name relative to its parent.
+   */
+  const std::string& GetName() const;
+
+  bool operator==(MechanismObject2d& rhs);

```

Comment:
This needs to be const and take a const parameter, `bool operator==(const MechanismObject2d&) const;`, or better, done as a non-member, non-friend function (outside the class), `bool operator==(const MechanismObject2d& a, const MechanismObject2d& b);`